readthedocs/proxito/middleware.py

```diff
@@ -172,32 +182,18 @@ def _set_request_attributes(self, request, unresolved_domain):
         """
         Set attributes in the request from the unresolved domain.
 
-        - If the project was extracted from the ``X-RTD-Slug`` header,
-          we set ``request.rtdheader`` to `True`.
-        - If the project was extracted from the public domain,
-          we set ``request.subdomain`` to `True`.
-        - If the project was extracted from a custom domain,
-          we set ``request.cname`` to `True`.
+        - Set ``request.unresolved_domain`` to the unresolved domain.
         - If the domain needs to redirect, set the canonicalize attribute accordingly.
```

Comment:
Only the `canonicalize` attributes are set here now, but I think we should be able to move these to the serve_docs view, since there is the only place we make use of them, maybe the http to https redirect could be done at the middleware level.

Reply:
Shouldn't we have a bigger object? I'd like to inject _one and only one_ object to the request with a known structure. As an example:

```python
class DataRequest:
  unresolved_domain
  canonicalize
  # ... others we will add in the future
```

What do you think?

Reply:
We are using these attributes in the docs view only, moving all the logic to the docs view serving should be easier

https://github.com/readthedocs/readthedocs.org/blob/f607db6243541100866ee7161ccd7a1bbbbc8f02/readthedocs/proxito/views/serve.py#L141-L160

Once that is moved, not sure if there is anything else that we will need to inject in the request, but if we have the need, I'd be fine creating another object.